CMakeLists.txt

```diff
@@ -4,13 +4,19 @@
 cmake_minimum_required (VERSION 3.15)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake-modules")
 
+# referenced options from other cmake code/files, avoid break if name ever change
+SET(CURL_TRANSPORT_OPT_NAME "BUILD_CURL_TRANSPORT")
+SET(WIN_TRANSPORT_OPT_NAME "BUILD_WIN_HTTP_TRANSPORT")
```

Comment:
See if cmake supports lists, so that you can pass it here (https://stackoverflow.com/questions/5248749/passing-a-list-to-a-cmake-macro?)
So that users can write `-DTRANSPORT="CURL WINHTTP MyOwnImpl"` (or whatever the cmake's format for lists is, i.e. maybe `"CURL;WINHTTP;MyOwnImpl"`).

Reply:
I am against the CMake option list.  I don't see a real benefit v/s having one option for each feature we are adding/removing.

Having the list requires us to validate the input is correct and ERROR is customer input some wrong arg in the list.
Customers need to look into Option description (or docs or error description) to learn about what are the options for the list. While having one option per transport would allow CMake discovery (Visual Studio and CMake UI) to quickly tell customer supported options to turn ON/OFF

Reply:
> I don't see a real benefit v/s having one option for each feature we are adding/removing.

- It is easier for the user to set and run. It won't require them to remember the exact names for multiple option.
- It is more scalable if we add a 3rd, 4th, ... nth transport layer.

> Customers need to look into Option description (or docs or error description) to learn about what are the options for the list. 

Right, but they only need to look at one option, which is easier to document.

> While having one option per transport would allow CMake discovery (Visual Studio and CMake UI) to quickly tell customer supported options to turn ON/OFF

I think it is possible to have the drop down work for multiple options, so they can be equally discoverable.

Reply:
Please feel free to create a PR after this to refactor and use a list, so you can figure it how to do the parsing appropriate selections.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/731